From ea2abed959bcc8942dcb491e085fb023f91260be Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Wed, 5 Apr 2017 14:51:35 -0700
Subject: [PATCH 2423/2748] HADOOP-14282. S3Guard: DynamoDBMetadata::prune()
 should self interrupt correctly. Contributed by
 Mingliang Liu

(cherry picked from commit dfeca6dd1ad5edbefc63f87ff86e2d5da83d5b86)

Change-Id: Idadf860b1f81823dd9bf7cdb9d7310525edb566c
---
 .../fs/s3a/s3guard/DynamoDBMetadataStore.java      |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java
index c3718d1..b2f011c 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java
@@ -651,7 +651,7 @@ public void prune(long modTime) throws IOException {
     int itemCount = 0;
     try {
       Collection<Path> deletionBatch =
-          new ArrayList(S3GUARD_DDB_BATCH_WRITE_REQUEST_LIMIT);
+          new ArrayList<>(S3GUARD_DDB_BATCH_WRITE_REQUEST_LIMIT);
       int delay = conf.getInt(S3GUARD_DDB_BACKGROUND_SLEEP_MSEC_KEY,
           S3GUARD_DDB_BACKGROUND_SLEEP_MSEC_DEFAULT);
       for (Item item : expiredFiles(modTime)) {
@@ -671,7 +671,7 @@ public void prune(long modTime) throws IOException {
         processBatchWriteRequest(pathToKey(deletionBatch), new Item[0]);
       }
     } catch (InterruptedException e) {
-      Thread.currentThread().interrupted();
+      Thread.currentThread().interrupt();
       throw new InterruptedIOException("Pruning was interrupted");
     }
     LOG.info("Finished pruning {} items in batches of {}", itemCount,
-- 
1.7.9.5

